src/tally_filter.F90

```diff
@@ -1584,17 +1592,23 @@ recursive subroutine find_offset(goal, univ, final, offset, path)
                     cycle
                   end if
 
-                  if (final >= lat % offset(map, k, l, m) + offset) then
+                  if (target_offset >= lat % offset(map, k, l, m) + offset) then
                     if (k == lat % n_rings .and. l == n_y .and. m == n_z) then
                       ! This is last lattice cell, so target must be here
                       lat_offset = lat % offset(map, k, l, m)
                       offset = offset + lat_offset
                       next_univ => universes(lat % universes(k, l, m))
-                      path = trim(path) // "(" // &
-                           trim(to_str(k - lat % n_rings)) // "," // &
-                           trim(to_str(l - lat % n_rings)) // "," // &
-                           trim(to_str(m)) // ")"
-                      call find_offset(goal, next_univ, final, offset, path)
+                      if (lat % is_3d) then
+                        path = trim(path) // "(" // &
+                             trim(to_str(k - lat % n_rings)) // "," // &
```

Comment:
Well, not quite zero-based indexing, rather (x,alpha) which ranges from -n_rings + 1 to n_rings as described [here](http://openmc.readthedocs.io/en/latest/methods/geometry.html#hexagonal-lattice-indexing).